x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -637,23 +637,23 @@ public boolean isIndexLifecycleAllowed() {
      *         {@code false}.
      */
     public boolean isEnrichAllowed() {
-        return isActive();
+        return allowForAllLicenses();
     }
 
     /**
      * Determine if EQL support should be enabled.
      * <p>
      *  EQL is available for all license types except {@link OperationMode#MISSING}
```

Comment:
We can remove the comment above, since it's not accurate (we don't check missing).

Reply:
Removed. I always went in and changed many of the inconsistent comments. However I am still not quite happy about current approach, e.g. "XXX is always allowed as long as there is an active license". This feels like documenting the internals instead of the overall purpose. It could be changed to "Determine whether XXX is allowed". But then it is basically duplicating the method name and might as well be removed. Please let me know how you think about this.

Reply:
My default position is that the existence of a comment should imply that something non-obvious is going on.

If the code is obvious (both in terms of what it tries to do, and why it does it that way), then the comment is a distraction. It's one more thing to read and understand that eventually just tells you what was right in front of you.

In these cases, I would hope the new structure is sufficient to make the comments unnecessary.

Reply:
Thanks Tim. This is my intention as well.